File: src/main/java/com/MyLibrary/library/model/Book.java
package com.MyLibrary.library.model;


import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

import javax.persistence.*;
import java.util.Date;
import java.util.List;
import java.util.Objects;
import java.util.UUID;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Book {

    @Id
    private UUID id;

    private String title;

    @NonNull
    @JoinColumn(name = "author_id")
    @ManyToOne(optional = false, fetch = FetchType.EAGER)
    private Author author;

    @JsonIgnore
    @OneToMany(mappedBy = "book", cascade = CascadeType.ALL)
    private List<Hire> hireList;

    private Date releaseDate;

    private boolean available = true;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Book book = (Book) o;
        return Objects.equals(id, book.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: frontend/myLibrary/src/app/shared/model/Book.ts
import {Author} from "@shared/model/Author";
import {Hire} from "@shared/model/Hire";

export interface Book {
  id: number;
  title: string;
  author: Author;
  hireList: Hire[];
  releaseDate: Date;
  available: boolean;
  hireDateFrom: Date;
  hireDateTo: Date;

}

export enum BookType{
  FANTASY = 'FANTASY',
  ROMANCE = 'ROMANCE',
  MYTHOLOGY = 'MYTHOLOGY',
  EDUCATION = 'EDUCATION'
}



